server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -2065,10 +2071,14 @@ public boolean restartNetwork(Long networkId, boolean cleanup, boolean makeRedun
     public boolean restartNetwork(RestartNetworkCmd cmd) throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         // This method restarts all network elements belonging to the network and re-applies all the rules
         Long networkId = cmd.getNetworkId();
+        NetworkVO network = getNetworkVO(cmd.getNetworkId(),"Nework to restart not found %s", String.valueOf(networkId));
         boolean cleanup = cmd.getCleanup();
+        if (network.getVpcId() != null && cleanup) {
+            throwInvalidIdException("Cannot restart a VPC tier with cleanup, please restart the whole VPC", network.getUuid(), "network tier");
```

Comment:
should throw `InvalidIdException` or another exception ?

Reply:
it does. it's a method call that throws `InvalidParameterException`.